query-engine/connector-test-kit-rs/query-engine-tests/tests/new/occ.rs

```diff
@@ -0,0 +1,389 @@
+        set.spawn(update_and_delete(runner.clone()));
+
+        while (set.join_next().await).is_some() {}
+
+        let res = find_one_resource(runner).await;
+
+        let expected = serde_json::json!({
+            "data": {
+            "findFirstResource": {
+              "occStamp": 1,
+              "id": 1
+            }
+          }
+        });
+
+        assert_eq!(res, expected);
+
+        Ok(())
+    }
+
+    #[connector_test(schema(occ_simple))]
+    async fn occ_delete_many_test(runner: Runner) -> TestResult<()> {
+        let runner = Arc::new(runner);
+
+        create_one_resource(runner.clone()).await;
+
+        let mut set = tokio::task::JoinSet::new();
+
+        set.spawn(delete_many_resource(runner.clone()));
+        set.spawn(delete_many_resource(runner.clone()));
+        set.spawn(delete_many_resource(runner.clone()));
+        set.spawn(delete_many_resource(runner.clone()));
+        set.spawn(delete_many_resource(runner.clone()));
+
+        let mut num_deleted: u64 = 0;
+        while let Some(res) = set.join_next().await {
+            if let Ok(row_count) = res {
+                if row_count > 0 {
+                    num_deleted += 1;
+                }
+            }
+        }
+
+        assert_eq!(num_deleted, 1);
+        let res = find_one_resource(runner).await;
+
+        let expected = serde_json::json!({
+            "data": {
+            "findFirstResource": serde_json::Value::Null
+          }
+        });
+        assert_eq!(res, expected);
+
+        Ok(())
+    }
+
+    // Because of the way upsert works this test is a little bit flaky. Ignoring until we fix upsert
+    #[allow(dead_code)]
+    #[ignore]
+    async fn occ_upsert_test(runner: Runner) -> TestResult<()> {
+        let runner = Arc::new(runner);
+
+        let mut set = tokio::task::JoinSet::new();
+
+        set.spawn(upsert_one_resource(runner.clone()));
+        set.spawn(upsert_one_resource(runner.clone()));
+        set.spawn(upsert_one_resource(runner.clone()));
+        set.spawn(upsert_one_resource(runner.clone()));
+        set.spawn(upsert_one_resource(runner.clone()));
+
+        while (set.join_next().await).is_some() {}
+
+        let res = find_one_resource(runner.clone()).await;
+
+        // MongoDB is different here and seems to only do one create with all the upserts
+        // where as all the sql databases will do one create and one upsert
+        let expected = if matches!(runner.connector(), ConnectorTag::MongoDb(_)) {
+            serde_json::json!({
+                "data": {
+                "findFirstResource": {
+                  "occStamp": 0,
+                  "id": 1
+                }
+              }
+            })
+        } else {
+            serde_json::json!({
+                "data": {
+                "findFirstResource": {
+                  "occStamp": 1,
+                  "id": 1
+                }
+              }
+            })
+        };
+        assert_eq!(res, expected);
```

Comment:
We have a macro for that. It called `match_connector_result`. Have a look at how it's used, it's pretty straightforward

Reply:
Yes but it doesn't compare json